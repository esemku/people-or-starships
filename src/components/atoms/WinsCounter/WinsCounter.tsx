import React from 'react';

type WinsCounterProps = {
  totalWins: number;
  className: string;
};

const WinsCounter: React.FC<WinsCounterProps> = ({
  totalWins,
  className,
  ...rest
}: WinsCounterProps) => {
  return (
    <div className={className} {...rest}>
      Total wins: {totalWins || 0}
    </div>
  );
};

export default WinsCounter;
