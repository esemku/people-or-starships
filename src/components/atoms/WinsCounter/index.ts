export { default as WinsCounter } from './WinsCounter';
