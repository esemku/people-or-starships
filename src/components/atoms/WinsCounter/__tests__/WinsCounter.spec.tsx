import React from 'react';
import { render, screen } from '@testing-library/react';

import WinsCounter from '../WinsCounter';

describe('<WinsCounter />', () => {
  it('renders correctly', () => {
    const { container } = render(
      <WinsCounter totalWins={1} className="className" />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders 0 for Total Wins when totalWins is undefined', () => {
    render(<WinsCounter totalWins={undefined} className="className" />);

    expect(screen.getByText('Total wins: 0').textContent).toContain(0);
  });
});
