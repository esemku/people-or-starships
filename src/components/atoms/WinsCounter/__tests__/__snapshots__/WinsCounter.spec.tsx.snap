// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WinsCounter /> renders correctly 1`] = `
<div>
  <div
    class="className"
  >
    Total wins: 
    1
  </div>
</div>
`;
